### PR TITLE
BUG #13760 - loader client de-duplicates batches of queries

### PR DESCRIPTION
**Pull request by sheldonkhall**

### CODE REVIEW

**Comment by sheldonkhall** [comment]
please retest this please

File: grakn-client/src/main/java/ai/grakn/client/BatchMutatorClient.java [hunk 1]
```diff
@@ -58,25 +58,26 @@
...
 
-    private static final Logger LOG = LoggerFactory.getLogger(LoaderClient.class);
+    private static final Logger LOG = LoggerFactory.getLogger(BatchMutatorClient.class);
 
     private final String POST = "http://%s" + TASKS;
     private final String GET = "http://%s" + TASKS + "/%s";
 
     private final Map<Integer,CompletableFuture> futures;
-    private final Collection<InsertQuery> queries;
+    private final Collection<Query> queries;
```
Comment (fppt) [inline grakn-client/src/main/java/ai/grakn/client/BatchMutatorClient.java]: Are Batch insertions different than normal insertions? Why the change in type?

File: grakn-client/src/main/java/ai/grakn/client/BatchMutatorClient.java [hunk 2]
```diff
@@ -87,14 +88,14 @@
...
+    public BatchMutatorClient(String keyspace, String uri) {
         this(keyspace, uri, (Json t) -> {});
     }
 
-    public LoaderClient(String keyspace, String uri, Consumer<Json> onCompletionOfTask){
+    public BatchMutatorClient(String keyspace, String uri, Consumer<Json> onCompletionOfTask){
         this.uri = uri;
         this.keyspace = keyspace;
-        this.queries = new HashSet<>();
+        this.queries = new ArrayList<>();
```
Comment (fppt) [inline grakn-client/src/main/java/ai/grakn/client/BatchMutatorClient.java]: Why? Is there any case where you want to execute the same insert query twice?

File: grakn-client/src/main/java/ai/grakn/client/BatchMutatorClient.java [hunk 3]
```diff
@@ -163,20 +164,24 @@ public LoaderClient setNumberActiveTasks(int size){
...
      * @param query insert query to be executed
      */
-    public void add(InsertQuery query){
-        queries.add(query);
-        if(queries.size() >= batchSize){
-            sendQueriesToLoader(new HashSet<>(queries));
-            queries.clear();
+    public void add(Query query){
+        if (query.isReadOnly()) {
+            throw new IllegalArgumentException(READ_ONLY_QUERY.getMessage(query.toString()));
```
Comment (fppt) [inline grakn-client/src/main/java/ai/grakn/client/BatchMutatorClient.java]: I might be wrong but I feel like this check may be redundant due to [1]

File: grakn-engine/src/main/java/ai/grakn/engine/loader/MutatorTask.java [hunk 4]
```diff
@@ -91,15 +93,21 @@ private boolean insertQueriesInOneTransaction(GraknGraph graph, Collection<Inser
...
-            return configuration.json().at(TASK_LOADER_INSERTS).asJsonList().stream()
+    private Collection<Query> getInserts(TaskConfiguration configuration){
+        if(configuration.json().has(TASK_LOADER_MUTATIONS)){
+            return configuration.json().at(TASK_LOADER_MUTATIONS).asJsonList().stream()
                     .map(Json::asString)
-                    .map(builder::<InsertQuery>parse)
+                    .map(builder::<Query<?>>parse)
+                    .map(query -> {
+                        if (query.isReadOnly()) {
+                            throw new IllegalArgumentException(READ_ONLY_QUERY.getMessage(query.toString()));
```
Comment (fppt) [inline grakn-engine/src/main/java/ai/grakn/engine/loader/MutatorTask.java]: [1] this
